helm/charts/oathkeeper/templates/service-api.yaml

```diff
@@ -21,7 +21,7 @@ spec:
     - port: {{ .Values.service.api.port }}
       targetPort: http-api
```

Comment:
Imho, if for Istio it is enough to have the naming convention on the service level (judging from [this](https://istio.io/latest/docs/reference/config/analysis/ist0118/)) I would refrain from modifications on deployment level, and rather change the naming only on ports.
Maybe we should go with the following:
- deployment port name is `foo`
- service port name is `{{ .Values.service.portPrefix}}-foo`
- so, instead of overriding the name, just add a specific prefix, which would default to http I guess